utils/src/main/kotlin/OrtProxySelector.kt

```diff
@@ -79,6 +79,12 @@ class OrtProxySelector(private val fallback: ProxySelector? = null) : ProxySelec
     private val proxyAuthentication = mutableMapOf<Proxy, PasswordAuthentication?>()
     private val proxyOrigins = mutableMapOf<String, MutableMap<String, MutableList<Proxy>>>()
 
+    private val noProxyUrls = Os.env["no_proxy"]?.let { list ->
+        list.split(',').mapNotNull { host ->
```

Comment:
I tried to find a specification for the format without any success. There seem to be conventions rather than a specification which are not all consistent. Can you please elaborate in the commit message about the format assumptions the code makes.

I found that several tools interpret an entry as a suffix, e.g. the entry `.bar.com` would match `foo.bar.com`.
Also several tools seem to support single star wildcared, e.g. `192.168.0.*` or `*.example.com`.
If these are not handled I believe a code comment would clarify things a bit.

Thus, also the variable `host` might be misleading.

Reply:
> There seem to be conventions rather than a specification which are not all consistent.

That's exactly my experience, too, also regarding the `http(s)_proxy` environment variables.

> Can you please elaborate in the commit message about the format assumptions the code makes.

Will do.

> Thus, also the variable `host` might be misleading.

But something like "somewhere.example.com" (which is supported) is still a host, no?

Reply:
> But something like "somewhere.example.com" (which is supported) is still a host, no?

I believe so.

Reply:
In the current version that uses string comparisons this variable name is gone anyway.

Reply:
Great!